### PR TITLE
Fix splitter errors for datasets without labels

### PR DESCRIPTION
**Pull request by Suzukazole**
## Description

Some datasets without labels such as USPTO, Swiss-Prot are not compatible with the splitters as the get_shard_size method looks for the length of y, which in this case is None. This is a crack at getting around this issue, I was able to load in and split the USPTO datasets with this. 

## Type of change

Please check the option that is related to your PR.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [X] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [X] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [X] Run `mypy -p deepchem` and check no errors
  - [X] Run `flake8 <modified file> --count` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by Suzukazole** [comment]
@rbharath  I feel there might be a better way to do this, but for now this works as intended with the USPTO loader.

File: deepchem/data/tests/test_shape.py [hunk 1]
```diff
@@ -106,3 +106,21 @@ def test_disk_dataset_get_legacy_shape_multishard():
...
+  num_datapoints = 100
+  num_features = 10
+  num_tasks = 10
+  # Generate data
+  X = np.random.rand(num_datapoints, num_features)
+  y = np.random.randint(2, size=(num_datapoints, num_tasks))
+  w = np.random.randint(2, size=(num_datapoints, num_tasks))
+  ids = np.array(["id"] * num_datapoints)
+
+  dataset = dc.data.DiskDataset.from_numpy(X, y, w, ids)
```
Comment (rbharath) [inline deepchem/data/tests/test_shape.py:122]: So here I'm thinking more if we do
```
dataset = dc.data.DiskDataset.from_numpy(X)
assert dataset.get_shard_size() == 100
```
The idea is we want to check that datasets without labels work correctly :)

Comment (Suzukazole) [reply deepchem/data/tests/test_shape.py:122]: yup! will add that

File: deepchem/data/tests/reaction_smiles.csv [hunk 2]
```diff
@@ -0,0 +1,5 @@
+reactions
```
Comment (rbharath) [inline deepchem/data/tests/reaction_smiles.csv:1]: Is this supposed to be in this PR?

File: deepchem/data/tests/test_shape.py [hunk 3]
```diff
@@ -106,3 +106,29 @@ def test_disk_dataset_get_legacy_shape_multishard():
...
+  The issue arises when attempting to load a dataset that does not have a labels
+  column. The create_dataset method of the DataLoader class sets the y to None
+  in this case, which causes the existing implementation of the get_shard_size()
+  method to fail, as it relies on the dataset having a not None y column. This
+  consequently breaks all methods depending on this, like the splitters for
+  example.
+
+  Note
+  ----
+  DiskDatasets without labels cannot be resharded!
```
Comment (rbharath) [inline deepchem/data/tests/test_shape.py:123]: This should be fixed! Can you raise a separate issue for this and we can get a fix going?

File: deepchem/data/tests/reaction_smiles.csv (hunk 2 above)
Comment (rbharath) [reply deepchem/data/tests/reaction_smiles.csv:1]: Oh I see this is used in the test.

**Review by rbharath — COMMENTED** [review]
For some reason I'm not seeing the latest CI run here.

If the CI is passing as expected we can go ahead and merge this PR in. I think code is all good. Let me know if I can merge

File: deepchem/data/tests/test_shape.py (hunk 3 above)
Comment (Suzukazole) [reply deepchem/data/tests/test_shape.py:123]: Yup will do!

**Comment by Suzukazole** [comment]
@rbharath Could you try restarting the checks maybe? I think I opened the PR when github actions was down.

**Review by rbharath — APPROVED** [review]
Ok I think this is good! Going to merge in

**Pull request merged**